### ISSUE TITLE
Calculate log_prob using resposne_length but not response_mask

### ISSUE THREAD

**Issue by yuzhaouoe** [volcengine/verl#2368]
Hi, I am training models with grpo in a multi-turn interaction setting. resposne_mask is useful to select the assistant's message to optimise, but I find the old_log_prob is calculated with response_length, which assumes the response is always at the end of the inptus:

`old_log_prob = self.actor_rollout_wg.compute_log_prob(batch)` [link](https://github.com/volcengine/verl/blob/c936ec7d5cebcc1d40f50296d28699696fbad6f5/verl/trainer/ppo/ray_trainer.py#L1194
)

`log_probs = full_log_probs.squeeze(-1)[:, -response_length - 1 : -1]  # (bsz, response_length)` [link](https://github.com/volcengine/verl/blob/c936ec7d5cebcc1d40f50296d28699696fbad6f5/verl/workers/actor/dp_actor.py#L239)

I wonder if my understanding is correct, and do I need to implement a new ppoactor to support multi-turn interaction training?

Thank you!

**Comment by wuxibin89** [comment]
https://github.com/volcengine/verl/blob/c936ec7d5cebcc1d40f50296d28699696fbad6f5/verl/workers/actor/dp_actor.py#L239
This `log_probs` contains LLM generated tokens and tool calling observation tokens, the observation tokens are masked  by response_mask in `compute_policy_loss`:
https://github.com/volcengine/verl/blob/c936ec7d5cebcc1d40f50296d28699696fbad6f5/verl/trainer/ppo/core_algos.py#L748

**Comment by yuzhaouoe** [comment]
> [verl/verl/workers/actor/dp_actor.py](https://github.com/volcengine/verl/blob/c936ec7d5cebcc1d40f50296d28699696fbad6f5/verl/workers/actor/dp_actor.py#L239)
> 
> Line 239 in [c936ec7](/volcengine/verl/commit/c936ec7d5cebcc1d40f50296d28699696fbad6f5)
> 
>  log_probs = full_log_probs.squeeze(-1)[:, -response_length - 1 : -1]  # (bsz, response_length) 
> 
> This `log_probs` contains LLM generated tokens and tool calling observation tokens, the observation tokens are masked by response_mask in `compute_policy_loss`:
> [verl/verl/trainer/ppo/core_algos.py](https://github.com/volcengine/verl/blob/c936ec7d5cebcc1d40f50296d28699696fbad6f5/verl/trainer/ppo/core_algos.py#L748)
> 
> Line 748 in [c936ec7](/volcengine/verl/commit/c936ec7d5cebcc1d40f50296d28699696fbad6f5)
> 
>  pg_loss = agg_loss(loss_mat=pg_losses, loss_mask=response_mask, loss_agg_mode=loss_agg_mode)

thank you for the explanation!
